cluster/charts/crossplane/templates/serviceaccount.yaml

```diff
@@ -1,7 +1,8 @@
+{{- if not .Values.serviceAccount.name -}}
```

Comment:
instead of relying on the name itself I'd add another boolean flag `.Values.serviceAccount.create`, it would allow to cover also the "I want the chart to create the serviceAccount, but it should have a specific name" use-case. You can see an example in the [kube-prometheus-stack helm chart](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/values.yaml), but it's pretty common practice across most well-known helm charts.

Reply:
done!

Reply:
I don't think I agree with this -- as @muvaf [pointed out](https://github.com/crossplane/crossplane/pull/2756#discussion_r766759008) in the first iteration of this PR, the presence of a name if enough for us to know whether to create or not.

Reply:
Sorry @hasheddan, I didn't go through the whole thread. I do agree it could be enough, but personally as a user I would have to read the template to be sure of its behavior, while the create flag is a common and recognizable pattern. Unfortunately helm charts are not easily configurable except for what the values provide out of the box, so not supporting a use-case usually leads users to either fork the chart or have to discard the chart.

Reply:
I agree with @phisco that the create flag is more explicit and makes the UX much more clear. However if you want me to just do it based on name, I can just revert the commit back to what it was initially both give the same output.

Reply:
@hasheddan has a decision been made on the implementation?